### PR TITLE
#766 creation of mandatory attribute in macroConnect element

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non regression tests were added (update of inputs, outputs, SA, algos)
- [ ] main documentation was updated (update of input/output file)
- [x] the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: etc/xsd/assembling.xsd [hunk 1]
```diff
@@ -125,5 +125,6 @@ SPDX-License-Identifier: MPL-2.0 -->
   <xs:complexType name="MacroConnect">
     <xs:attribute name="macroConnection" use="required" type="xs:string"/>
     <xs:attribute name="id" use="required" type="xs:string"/>
+    <xs:attribute name="mandatory" type="xs:boolean"/>
```
Comment (gaouizer) [inline etc/xsd/assembling.xsd:128]: I am wondering if the `use="optional"` attribute would be necessary in that case?

**Review by gaouizer — CHANGES_REQUESTED** [review]
three general comments:

- wouldn't be necessary to edit `DynModelFilterAlgorithm::filterPartiallyConnectedDynamicModels` in order to prevent partially connected automaton deletion?
- adding tests is necessary to make sure that these pieces of code work as expected  
- Linux build failure in the CI seems to be related to doxygen, this needs to be investigated

**Comment by barondim** [comment]
- [ ] I made a first draft for this task but we need to confirm if it was the work that was expected
- [ ] Addition of unit tests

File: etc/xsd/assembling.xsd (hunk 1 above)
Comment (rosiereflo) [reply etc/xsd/assembling.xsd:128]: Hi, yes, the default value would be "false"

File: sources/Algo/src/DynModelFilterAlgorithm.cpp [hunk 2]
```diff
@@ -110,13 +110,17 @@ DynModelFilterAlgorithm::filterPartiallyConnectedDynamicModels() {
       continue;
 
     for (const inputs::AssemblingDataBase::MacroConnect& macroConnect : automaton.second.macroConnects) {
-      auto found = std::find_if(
+      if (!macroConnect.mandatory) {
```
Comment (rosiereflo) [inline sources/Algo/src/DynModelFilterAlgorithm.cpp:113]: this is the opoosite: if the connection is mandatory, perform the check

**Comment by rosiereflo** [comment]
looks ok now, only ut and nrt are missing

File: sources/Inputs/include/AssemblingDataBase.h [hunk 3]
```diff
@@ -100,8 +100,14 @@ class AssemblingDataBase {
    * @brief Macro connect XML element defining a macro connection of a dynamic model
    */
   struct MacroConnect {
+    /**
+     * @brief Default constructor
+     */
+    MacroConnect() : mandatory(false) {}
```
Comment (rosiereflo) [inline sources/Inputs/include/AssemblingDataBase.h:106]: default is true

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Review by rosiereflo — APPROVED** [review]

**Review by gaouizer — APPROVED** [review]

**Pull request merged**